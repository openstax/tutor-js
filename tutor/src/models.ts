// This file controls the order of imports in order to prevent circular dependencies
// NO code should directly import from a model file, instead everything should import from here.
// For more info on this technique:
// https://medium.com/visual-development/how-to-fix-nasty-circular-dependency-issues-once-and-for-all-in-javascript-typescript-a04c987cf0de

// export * from './models/toasts'

export * from './models/types'

import Bignum  from 'shared/model/bignum';
import Time, { Interval } from 'shared/model/time';
import { Toast, currentToasts } from 'shared/model/toasts'

export { Time, Interval, Bignum, Toast, currentToasts }

export * from './models/feature_flags'

export * from './models/window-scroll'
export * from './models/window-size'

export * from './models/appearance_codes'
export * from './models/chapter-section'
export * from './models/related-content'

export * from './models/reference-book/node'
export * from './models/reference-book'

export * from './models/ecosystems/book'
export * from './models/ecosystems/ecosystem'
export * from './models/ecosystems'

export * from './models/practice-questions'
export * from './models/grading/templates'
export * from './models/notes/note'
export * from './models/notes/ux'
export * from './models/notes'

export * from './models/exercises/exercise'
export * from './models/exercises'

export * from './models/student-tasks/student'
export * from './models/student-tasks/step'
export * from './models/student-tasks/info-step'
export * from './models/student-tasks/step-group'
export * from './models/student-tasks/task'
export * from './models/student-tasks'

export * from './models/practice-questions'
export * from './models/practice-questions/practice-question'

export * from './models/task-plans/teacher/dropped_question'
export * from './models/task-plans/teacher/tasking'
export * from './models/task-plans/teacher/plan'
export * from './models/task-plans/teacher/past'
export * from './models/task-plans/teacher/grade'
export * from './models/task-plans/teacher/scores'
export * from './models/task-plans/teacher'
export * from './models/task-plans/teacher/stats'

export * from './models/task-plans/student/task'
export * from './models/task-plans/student'

export * from './models/course/term'
// export * from './models/course/enroll'
export * from './models/course/create'
export * from './models/course/offerings/offering'
export * from './models/course/offerings'
export * from './models/course/information'
export * from './models/course/teacher-profile'
export * from './models/course/teacher'
export * from './models/course/student'
export * from './models/course/ux'
export * from './models/course/roster'
export * from './models/course/role'
export * from './models/course/period'
export * from './models/course/lms'
export * from './models/course/pair-to-lms'
export * from './models/course'

export * from './models/courses-map'

export * from './models/tour/actions'
export * from './models/tour/step'
export * from './models/tour/region'
export * from './models/tour/context'
export * from './models/tour/ride'
export * from './models/tour'

export * from './models/user/terms'
export * from './models/user'
export * from './models/user/menu'

export * from './models/research-surveys/survey'
export * from './models/research-surveys'

export * from './models/courses-map'

export * from './models/exercises'
export * from './models/exercises'
export * from './models/grading/templates'
export * from './models/job'
export * from './models/jobs/task-plan-publish'
export * from './models/jobs/lms-score-push'
export * from './models/jobs/scores-export'
export * from './models/loader'


export * from './models/practice-questions'

export * from './models/purchases/purchase'
export * from './models/purchases'

export * from './models/research-surveys'

export * from './models/response_validation'

export * from './models/subject-order'

export * from './models/scores/task-result'
export * from './models/scores/heading'
export * from './models/scores/student'
export * from './models/scores/period'
export * from './models/scores'


export * from './models/app/pulse-insights'
export * from './models/app/raven'

// export * from './models/reference-book/ux'

//export * from './models/app'
