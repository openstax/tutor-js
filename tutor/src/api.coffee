# This file manages all async state transitions.
#
# These attach to actions to help state changes along.
#
# For example, `TaskActions.load` everntually yields either
# `TaskActions.loaded` or `TaskActions.FAILED`
_ = require 'underscore'

{apiHelper, IS_LOCAL, toParams, onRequestError, route} = require './helpers/api'
TimeHelper = require './helpers/time'
{CurrentUserActions, CurrentUserStore} = require './flux/current-user'
{CourseActions} = require './flux/course'
{CoursePracticeActions} = require './flux/practice'
{CourseGuideActions} = require './flux/guide'
{JobActions} = require './flux/job'
{EcosystemsActions} = require './flux/ecosystems'
PerformanceForecast = require './flux/performance-forecast'

{ScoresActions} = require './flux/scores'
{ScoresExportActions} = require './flux/scores-export'
{RosterActions, RosterStore} = require './flux/roster'
{PeriodActions} = require './flux/period'

{TaskActions} = require './flux/task'
{TaskPanelActions, TaskPanelStore} = require './flux/task-panel'
{TaskStepActions} = require './flux/task-step'
{ StudentIdStore, StudentIdActions } = require './flux/student-id'
{TaskPlanActions, TaskPlanStore} = require './flux/task-plan'
{TaskTeacherReviewActions, TaskTeacherReviewStore} = require './flux/task-teacher-review'
{TaskPlanStatsActions, TaskPlanStatsStore} = require './flux/task-plan-stats'

{TocActions} = require './flux/toc'
{ExerciseActions, ExerciseStore} = require './flux/exercise'
{TeacherTaskPlanActions, TeacherTaskPlanStore} = require './flux/teacher-task-plan'
{StudentDashboardActions} = require './flux/student-dashboard'
{CourseListingActions, CourseListingStore} = require './flux/course-listing'
{CCDashboardStore, CCDashboardActions} = require './flux/cc-dashboard'

{ReferenceBookActions, ReferenceBookStore} = require './flux/reference-book'
{ReferenceBookPageActions, ReferenceBookPageStore} = require './flux/reference-book-page'
{ReferenceBookExerciseActions, ReferenceBookExerciseStore} = require './flux/reference-book-exercise'

{NotificationActions, NotificationStore} = require './flux/notifications'

SharedNetworking = require 'shared/src/model/networking'

BOOTSTRAPED_STORES = {
  user:   CurrentUserActions.loaded
  courses: CourseListingActions.loaded
}

start = (bootstrapData) ->
  for storeId, action of BOOTSTRAPED_STORES
    data = bootstrapData[storeId]
    action(data) if data

  SharedNetworking.onError(onRequestError)

  apiHelper TaskActions, TaskActions.load, TaskActions.loaded, 'GET', (id) ->
    url: "/api/tasks/#{id}"

  # apiHelper TaskActions, TaskActions.save, TaskActions.saved, 'PATCH', (id, obj) ->
  #   url: "/api/tasks/#{id}"
  #   payload: obj

  saveHelper = (id, courseId) ->
    obj = TaskPlanStore.getChanged(id)
    if TaskPlanStore.isNew(id)
      url: "/api/courses/#{courseId}/plans"
      httpMethod: 'POST'
      payload: obj
    else
      url: "/api/plans/#{id}"
      httpMethod: 'PATCH'
      payload: obj

  apiHelper TaskPlanActions, TaskPlanActions.save, TaskPlanActions.saved, null, saveHelper

  apiHelper TaskPlanActions, TaskPlanActions.delete, TaskPlanActions.deleted, 'DELETE', saveHelper

  apiHelper TaskPlanActions, TaskPlanActions.load , TaskPlanActions.loaded, 'GET', (id) ->
    url: "/api/plans/#{id}"

  apiHelper TaskPlanStatsActions, TaskPlanStatsActions.load , TaskPlanStatsActions.loaded, 'GET', (id) ->
    url: "/api/plans/#{id}/stats"

  # Note: the below exercise endpoints share the same store.
  # The contents of the json payload is identical, except the second includes an is_excluded flag
  # since it operates at the course level
  #
  # This one loads using an ecosystemId
  apiHelper ExerciseActions, ExerciseActions.loadForEcosystem,
    ExerciseActions.loadedForEcosystem, 'GET', (ecosystemId, page_ids, requestType = 'homework_core') ->
      url: "/api/ecosystems/#{ecosystemId}/exercises/#{requestType}?#{toParams({page_ids})}"
  # And this one loads using a courseId
  apiHelper ExerciseActions, ExerciseActions.loadForCourse,
    ExerciseActions.loadedForCourse, 'GET', (courseId, pageIds, ecosystemId = null, requestType = 'homework_core') ->
      params = { page_ids: pageIds }
      params['ecosystem_id'] = ecosystemId if ecosystemId?
      url: "/api/courses/#{courseId}/exercises/#{requestType}?#{toParams(params)}"

  apiHelper ExerciseActions, ExerciseActions.saveExerciseExclusion,
    ExerciseActions.exclusionsSaved, 'PUT', (courseId) ->
      url: "/api/courses/#{courseId}/exercises"
      payload: _.map ExerciseStore.getUnsavedExclusions(), (is_excluded, id) -> {id, is_excluded}

  apiHelper TocActions, TocActions.load, TocActions.loaded, 'GET', (ecosystemId) ->
    url: "/api/ecosystems/#{ecosystemId}/readings"

  apiHelper CourseGuideActions, CourseGuideActions.load, CourseGuideActions.loaded, 'GET', (courseId) ->
    url: "/api/courses/#{courseId}/guide"

  apiHelper CourseActions, CourseActions.save, CourseActions.saved, 'PATCH', (courseId, params) ->
    url: "/api/courses/#{courseId}"
    payload: params

  apiHelper CCDashboardActions, CCDashboardActions.load, CCDashboardActions.loaded, 'GET', (courseId) ->
    url: "/api/courses/#{courseId}/cc/dashboard"

  apiHelper CoursePracticeActions, CoursePracticeActions.create, CoursePracticeActions.created, 'POST', (courseId, params) ->
    url: "/api/courses/#{courseId}/practice"
    payload: params

  apiHelper CoursePracticeActions, CoursePracticeActions.createSilent, CoursePracticeActions.created, 'POST', (courseId, params) ->
    url: "/api/courses/#{courseId}/practice"
    payload: params
  , displayError: false

  apiHelper CoursePracticeActions, CoursePracticeActions.load, CoursePracticeActions.loaded, 'GET', (courseId) ->
    url: "/api/courses/#{courseId}/practice"

  apiHelper CourseActions, CourseActions.load, CourseActions.loaded, 'GET', (courseId) ->
    url: "/api/courses/#{courseId}"

  apiHelper PerformanceForecast.Student.actions, PerformanceForecast.Student.actions.load,
    PerformanceForecast.Student.actions.loaded, 'GET', (id) -> url: "/api/courses/#{id}/guide"

  apiHelper PerformanceForecast.Teacher.actions, PerformanceForecast.Teacher.actions.load,
    PerformanceForecast.Teacher.actions.loaded, 'GET', (id) -> url: "/api/courses/#{id}/teacher_guide"

  apiHelper PerformanceForecast.TeacherStudent.actions, PerformanceForecast.TeacherStudent.actions.load,
    PerformanceForecast.TeacherStudent.actions.loaded, 'GET', (id, {roleId}) ->
      url: "/api/courses/#{id}/guide/role/#{roleId}"

  apiHelper ScoresActions, ScoresActions.load, ScoresActions.loaded, 'GET', (id) ->
    url: "/api/courses/#{id}/performance"
  apiHelper ScoresActions, ScoresActions.acceptLate, ScoresActions.acceptedLate, 'PUT', (id) ->
    url: "/api/tasks/#{id}/accept_late_work"
  apiHelper ScoresActions, ScoresActions.rejectLate, ScoresActions.rejectedLate, 'PUT', (id) ->
    url: "/api/tasks/#{id}/reject_late_work"
  apiHelper StudentDashboardActions, StudentDashboardActions.hide, StudentDashboardActions.hidden, 'DELETE', (id) ->
    url: "/api/tasks/#{id}"


  apiHelper ScoresExportActions, ScoresExportActions.load, ScoresExportActions.loaded, 'GET', (id) ->
    url: "/api/courses/#{id}/performance/exports"

  apiHelper ScoresExportActions, ScoresExportActions.export, ScoresExportActions.exported, 'POST', (id) ->
    url: "/api/courses/#{id}/performance/export"

  apiHelper JobActions, JobActions.load, JobActions.loaded, 'GET', (id) ->
    url: "/api/jobs/#{id}"
  , displayError: false

  apiHelper EcosystemsActions, EcosystemsActions.load, EcosystemsActions.loaded, 'GET', ->
    url: "/api/ecosystems"

  apiHelper RosterActions, RosterActions.teacherDelete, RosterActions.teacherDeleted, 'DELETE', (id) ->
    url: "/api/teachers/#{id}"

  apiHelper RosterActions, RosterActions.delete, RosterActions.deleted, 'DELETE', (id) ->
    url: "/api/students/#{id}"

  apiHelper RosterActions, RosterActions.save, RosterActions.saved, 'PATCH', (id, params) ->
    url: "/api/students/#{id}", payload: params

  route 'PUT', "/api/students/{studentId}/undrop",
    actions: RosterActions, trigger: 'undrop', onSuccess: 'undropped'
    errorHandlers:
      already_active: 'onUndropAlreadyActive'
      student_identifier_has_already_been_taken: 'recordDuplicateStudentIdError'

  route 'PATCH', '/api/students/{studentId}',
    actions: RosterActions, trigger: 'saveStudentIdentifier', onSuccess: 'savedStudentIdentifier',
    payload: ({courseId, studentId}) ->
      student_identifier: RosterStore.getStudentIdentifier(courseId, studentId)
    errorHandlers:
      student_identifier_has_already_been_taken: 'recordDuplicateStudentIdError'

  apiHelper RosterActions, RosterActions.create, RosterActions.created, 'POST', (courseId, params) ->
    url: "/api/courses/#{courseId}/roster", payload: params
  apiHelper RosterActions, RosterActions.load, RosterActions.loaded, 'GET', (id) ->
    url: "/api/courses/#{id}/roster"

  apiHelper StudentIdActions, StudentIdActions.save, StudentIdActions.saved, 'PATCH', (courseId, params) ->
    url: "/api/user/courses/#{courseId}/student",
    payload: params,

  , displayError: false, handleError: StudentIdActions.errored

  apiHelper PeriodActions, PeriodActions.create, PeriodActions.created, 'POST', (courseId, params) ->
    url: "/api/courses/#{courseId}/periods", payload: params
  apiHelper PeriodActions, PeriodActions.save, PeriodActions.saved, 'PATCH', (id, period, params) ->
    url: "/api/periods/#{period}", payload: params
  apiHelper PeriodActions, PeriodActions.delete, PeriodActions.deleted, 'DELETE', (id) ->
    url: "/api/periods/#{id}"
  apiHelper PeriodActions, PeriodActions.restore, PeriodActions.restored, 'PUT', (id) ->
    url: "/api/periods/#{id}/restore"

  apiHelper TaskStepActions, TaskStepActions.load, TaskStepActions.loaded, 'GET', (id) ->
    throw new Error('BUG: Wrong type') unless typeof id is 'string' or typeof id is 'number'
    url: "/api/steps/#{id}"

  # Loading personalized step should fail without should server error modal.
  apiHelper TaskStepActions, TaskStepActions.loadPersonalized, TaskStepActions.loaded, 'GET', (id) ->
    throw new Error('BUG: Wrong type') unless typeof id is 'string' or typeof id is 'number'
    url: "/api/steps/#{id}"
  , displayError: false, handleError: TaskStepActions.loadedNoPersonalized

  # # Go from complete to load so we fetch the new JSON
  # apiHelper TaskStepActions, TaskStepActions.complete, TaskStepActions.loaded, 'PUT', (id) ->
  #   url: "/api/steps/#{id}/completed"

  # Go from complete to load so we fetch the new JSON
  apiHelper TaskStepActions, TaskStepActions.complete, TaskStepActions.completed, 'PUT', (id) ->
    url: "/api/steps/#{id}/completed"

  apiHelper TaskStepActions, TaskStepActions.loadRecovery, TaskStepActions.loadedRecovery, 'PUT', (id) ->
    url: "/api/steps/#{id}/recovery"

  apiHelper TaskStepActions, TaskStepActions.setFreeResponseAnswer, TaskStepActions.saved, 'PATCH', (id, free_response) ->
    url: "/api/steps/#{id}"
    payload: {free_response}

  apiHelper TaskStepActions, TaskStepActions.setAnswerId, TaskStepActions.saved, 'PATCH', (id, answer_id) ->
    url: "/api/steps/#{id}"
    payload: {answer_id}

  apiHelper TaskTeacherReviewActions, TaskTeacherReviewActions.load, TaskTeacherReviewActions.loaded, 'GET', (id) ->
    url: "/api/plans/#{id}/review"

  apiHelper TeacherTaskPlanActions, TeacherTaskPlanActions.load, TeacherTaskPlanActions.loaded, 'GET',
    (courseId, start_at, end_at) ->
      params = toParams({start_at, end_at})

      url: "/api/courses/#{courseId}/dashboard?#{params}"

  apiHelper CurrentUserActions, CurrentUserActions.load, CurrentUserActions.loaded, 'GET', ->
    url: '/api/user'

  apiHelper CourseListingActions, CourseListingActions.load, CourseListingActions.loaded, 'GET', ->
    url: '/api/user/courses'

  apiHelper ReferenceBookActions, ReferenceBookActions.load, ReferenceBookActions.loaded, 'GET', (ecosystemId) ->
    url: "/api/ecosystems/#{ecosystemId}/readings"


  apiHelper ReferenceBookPageActions, ReferenceBookPageActions.load, ReferenceBookPageActions.loaded, 'GET', (cnxId) ->
    url: "/api/pages/#{cnxId}"

  apiHelper ReferenceBookPageActions, ReferenceBookPageActions.loadSilent, ReferenceBookPageActions.loaded, 'GET', (cnxId) ->
    url: "/api/pages/#{cnxId}"
  , displayError: false

  apiHelper ReferenceBookExerciseActions,
    ReferenceBookExerciseActions.load,
    ReferenceBookExerciseActions.loaded,
    'GET',
    (exerciseAPIUrl) ->
      url: exerciseAPIUrl

  apiHelper StudentDashboardActions, StudentDashboardActions.load, StudentDashboardActions.loaded, 'GET', (courseId) ->
    url: "/api/courses/#{courseId}/dashboard"

  apiHelper NotificationActions, NotificationActions.loadUpdates, NotificationActions.loadedUpdates, 'GET', ->
    url: "/api/notifications"
  , displayError: false

  CurrentUserActions.logout.addListener 'trigger', ->
    # Logging out programatically needs to be done via a form submission or follow redirects
    # $.ajax('/accounts/logout', {method: 'DELETE'})
    # .always ->
    console.warn('Logging out programatically needs to be done via a form submission or follow redirects')
    window.location.href = '/'


module.exports = {start}
