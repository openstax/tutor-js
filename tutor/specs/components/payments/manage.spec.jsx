import { C, ApiMock, TimeMock } from '../../helpers';
import Payments from '../../../src/components/payments/manage';
import renderer from 'react-test-renderer';
import { currentPurchases } from '../../../src/models';
import Router from '../../../src/helpers/router';
import mockData from '../../../api/purchases.json';

jest.mock('../../../src/helpers/router');

describe('Student Payments Management', () => {

    TimeMock.setTo(new Date('2017-07-01'));

    ApiMock.intercept({
        'purchases': mockData,
    })

    beforeEach(() => {
        Router.currentParams.mockReturnValue({ courseId: '2' });
        Router.makePathname.mockImplementation(() => '/foo');
    });

    it('renders and matches snapshot', async () => {
        const component = renderer.create(<C noRef><Payments /></C>)

        // clearing react lifecycle
        await new Promise(resolve => setTimeout(resolve, 1));
        await new Promise(resolve => setTimeout(resolve, 1));

        expect.snapshot(component).toMatchSnapshot();
        component.unmount()
    });

    it('renders empty when no payments', () => {
        currentPurchases.clear();
        expect.snapshot(<C noRef><Payments /></C>).toMatchSnapshot();
    });

});
