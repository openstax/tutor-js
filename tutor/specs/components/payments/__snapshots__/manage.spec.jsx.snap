// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Student Payments Management renders and matches snapshot 1`] = `
.c2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  min-height: 55px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 25px 0 10px;
  border-bottom: 1px solid #d5d5d5;
  background-color: white;
}

.c0 > div {
  margin-bottom: 12px;
}

.c0 .sticky-table {
  margin-left: 15px;
  margin-right: 15px;
}

.c1 {
  width: 100%;
  color: #027EB5;
  padding-left: 10px;
}

.c3 {
  font-size: 3.5rem;
  padding-left: 12px;
}

.c4 {
  padding: 4rem 10rem 50rem;
  background-color: white;
  max-width: 1000px;
  margin: 0 auto;
}

.c4 table {
  font-size: 1.4rem;
}

.c4 table thead th,
.c4 table tbody td {
  padding: 1rem;
}

.c4 table tbody td {
  vertical-align: middle;
}

.c4 table tbody .refunded .total {
  color: #489200;
}

.c4 .footer {
  margin: 10px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 .footer > * {
  margin-top: 10px;
}

@media (max-width:600px) {
  .c0 {
    padding-bottom: 0;
  }
}

@media (max-width:600px) {
  .c0 .sticky-table {
    margin-left: 0;
    margin-right: 0;
    margin-bottom: 0;
  }
}

@media (max-width:999px) {
  .c4 {
    padding: 0;
  }

  .c4 .footer {
    padding: 1rem 24px;
  }
}

@media (max-width:600px) {
  .c4 {
    padding: 0;
  }

  .c4 .footer {
    padding: 1rem 8px;
  }
}

<div
  className="openstax-debug-content"
>
  <div
    data-purpose="tour-conductor-wrapper"
  >
    <div>
      <div
        className="c0 task-homework breadcrumbs-wrapper"
        role="dialog"
        tabIndex="-1"
      >
        <div
          className="c1"
        >
          <a
            href="/foo"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-angle-left fa-w-8 fa-lg c2 ox-icon ox-icon-angle-left -move-exercise-up circle"
              data-icon="angle-left"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 256 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            My Courses
          </a>
        </div>
        <h1
          className="c3"
        >
          Manage Payments
        </h1>
      </div>
      <div
        className="c4 manage-payments"
      >
        <table
          className="table table-striped"
        >
          <thead>
            <tr>
              <th>
                Item
              </th>
              <th>
                Transaction date
              </th>
              <th>
                Order number
              </th>
              <th>
                Amount
              </th>
              <th>
                Invoice
              </th>
              <th />
            </tr>
          </thead>
          <tbody>
            <tr
              className="refunded"
            >
              <td>
                unit test product refund
              </td>
              <td>
                Jul 19th 2017
              </td>
              <td>
                cufu6hr4xicy:refund
              </td>
              <td
                className="total"
              >
                -1591.48
              </td>
              <td />
              <td
                className="refund"
              />
            </tr>
            <tr
              className=""
            >
              <td>
                unit test product
              </td>
              <td>
                Jul 16th 2017
              </td>
              <td>
                cufu6hr4xicy
              </td>
              <td
                className="total"
              >
                1591.48
              </td>
              <td>
                <a
                  className="invoice-link"
                  data-identifier="cufu6hr4xicy"
                  href="url-for-test/invoice/cufu6hr4xicy"
                  onClick={[Function]}
                >
                  Invoice
                </a>
              </td>
              <td
                className="refund"
              />
            </tr>
            <tr
              className="refunded"
            >
              <td>
                unit test product refund
              </td>
              <td>
                Jul 18th 2017
              </td>
              <td>
                kjxdvs9wd7rw:refund
              </td>
              <td
                className="total"
              >
                -1388.07
              </td>
              <td />
              <td
                className="refund"
              />
            </tr>
            <tr
              className=""
            >
              <td>
                unit test product
              </td>
              <td>
                Jul 16th 2017
              </td>
              <td>
                kjxdvs9wd7rw
              </td>
              <td
                className="total"
              >
                1388.07
              </td>
              <td>
                <a
                  className="invoice-link"
                  data-identifier="kjxdvs9wd7rw"
                  href="url-for-test/invoice/kjxdvs9wd7rw"
                  onClick={[Function]}
                >
                  Invoice
                </a>
              </td>
              <td
                className="refund"
              />
            </tr>
            <tr
              className=""
            >
              <td>
                unit test product
              </td>
              <td>
                Jul 16th 2017
              </td>
              <td>
                3dmy8nu4v4bg
              </td>
              <td
                className="total"
              >
                357.25
              </td>
              <td>
                <a
                  className="invoice-link"
                  data-identifier="3dmy8nu4v4bg"
                  href="url-for-test/invoice/3dmy8nu4v4bg"
                  onClick={[Function]}
                >
                  Invoice
                </a>
              </td>
              <td
                className="refund"
              >
                <button
                  className="async-button btn btn-primary"
                  data-identifier="3dmy8nu4v4bg"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Request Refund
                </button>
              </td>
            </tr>
            <tr
              className=""
            >
              <td>
                OpenStax Tutor
              </td>
              <td>
                Jul 14th 2017
              </td>
              <td>
                fxa4s2kj75fr
              </td>
              <td
                className="total"
              >
                13.41
              </td>
              <td>
                <a
                  className="invoice-link"
                  data-identifier="fxa4s2kj75fr"
                  href="url-for-test/invoice/fxa4s2kj75fr"
                  onClick={[Function]}
                >
                  Invoice
                </a>
              </td>
              <td
                className="refund"
              >
                <button
                  className="async-button btn btn-primary"
                  data-identifier="fxa4s2kj75fr"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Request Refund
                </button>
              </td>
            </tr>
            <tr
              className=""
            >
              <td>
                OpenStax Tutor
              </td>
              <td>
                Jul 13th 2017
              </td>
              <td>
                ru6ifviicwki
              </td>
              <td
                className="total"
              >
                13.41
              </td>
              <td>
                <a
                  className="invoice-link"
                  data-identifier="ru6ifviicwki"
                  href="url-for-test/invoice/ru6ifviicwki"
                  onClick={[Function]}
                >
                  Invoice
                </a>
              </td>
              <td
                className="refund"
              >
                <button
                  className="async-button btn btn-primary"
                  data-identifier="ru6ifviicwki"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Request Refund
                </button>
              </td>
            </tr>
            <tr
              className=""
            >
              <td>
                OpenStax Tutor
              </td>
              <td>
                Jul 13th 2017
              </td>
              <td>
                nuvwqun23ztq
              </td>
              <td
                className="total"
              >
                13.41
              </td>
              <td>
                <a
                  className="invoice-link"
                  data-identifier="nuvwqun23ztq"
                  href="url-for-test/invoice/nuvwqun23ztq"
                  onClick={[Function]}
                >
                  Invoice
                </a>
              </td>
              <td
                className="refund"
              >
                <button
                  className="async-button btn btn-primary"
                  data-identifier="nuvwqun23ztq"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Request Refund
                </button>
              </td>
            </tr>
            <tr
              className=""
            >
              <td>
                OpenStax Tutor
              </td>
              <td>
                Jul 13th 2017
              </td>
              <td>
                33uqbxbi324n
              </td>
              <td
                className="total"
              >
                13.41
              </td>
              <td>
                <a
                  className="invoice-link"
                  data-identifier="33uqbxbi324n"
                  href="url-for-test/invoice/33uqbxbi324n"
                  onClick={[Function]}
                >
                  Invoice
                </a>
              </td>
              <td
                className="refund"
              >
                <button
                  className="async-button btn btn-primary"
                  data-identifier="33uqbxbi324n"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Request Refund
                </button>
              </td>
            </tr>
          </tbody>
        </table>
        <div
          className="footer"
        >
          <a
            className="refund-policy external-icon"
            href="https://openstax.secure.force.com/help/articles/FAQ/OpenStax-Tutor-Beta-Student-Refund-Policy"
            tabIndex={0}
            target="_blank"
          >
            Refund policy for OpenStax Tutor Beta courses
          </a>
          <div
            className="help"
          >
            Need help? 
            <a
              href="mailto:support@openstax.org"
            >
              Contact Support
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="visible-when-debugging"
    >
      <div
        className="tour-spy-info"
      >
        available regions: []; region tour ids: []; audience tags: []; tour tags: []; eligible tours: []; TOUR RIDE: &lt;none&gt;
      </div>
    </div>
  </div>
  <a
    aria-hidden="true"
    className="debug-toggle-link"
    href="#spy"
    onClick={[Function]}
    tabIndex={-1}
  >
    π
  </a>
</div>
`;

exports[`Student Payments Management renders empty when no payments 1`] = `
.c2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  min-height: 55px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 25px 0 10px;
  border-bottom: 1px solid #d5d5d5;
  background-color: white;
}

.c0 > div {
  margin-bottom: 12px;
}

.c0 .sticky-table {
  margin-left: 15px;
  margin-right: 15px;
}

.c1 {
  width: 100%;
  color: #027EB5;
  padding-left: 10px;
}

.c3 {
  font-size: 3.5rem;
  padding-left: 12px;
}

.c4 {
  padding: 4rem 10rem 50rem;
  background-color: white;
  max-width: 1000px;
  margin: 0 auto;
}

.c4 table {
  font-size: 1.4rem;
}

.c4 table thead th,
.c4 table tbody td {
  padding: 1rem;
}

.c4 table tbody td {
  vertical-align: middle;
}

.c4 table tbody .refunded .total {
  color: #489200;
}

.c4 .footer {
  margin: 10px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 .footer > * {
  margin-top: 10px;
}

@media (max-width:600px) {
  .c0 {
    padding-bottom: 0;
  }
}

@media (max-width:600px) {
  .c0 .sticky-table {
    margin-left: 0;
    margin-right: 0;
    margin-bottom: 0;
  }
}

@media (max-width:999px) {
  .c4 {
    padding: 0;
  }

  .c4 .footer {
    padding: 1rem 24px;
  }
}

@media (max-width:600px) {
  .c4 {
    padding: 0;
  }

  .c4 .footer {
    padding: 1rem 8px;
  }
}

<div
  className="openstax-debug-content"
>
  <div
    data-purpose="tour-conductor-wrapper"
  >
    <div>
      <div
        className="c0 task-homework breadcrumbs-wrapper"
        role="dialog"
        tabIndex="-1"
      >
        <div
          className="c1"
        >
          <a
            href="/foo"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-angle-left fa-w-8 fa-lg c2 ox-icon ox-icon-angle-left -move-exercise-up circle"
              data-icon="angle-left"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 256 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            My Courses
          </a>
        </div>
        <h1
          className="c3"
        >
          Manage Payments
        </h1>
      </div>
      <div
        className="c4 manage-payments"
      >
        <div
          className="empty"
        >
          <h3>
            No payments were found for your account.
          </h3>
        </div>
        <div
          className="footer"
        >
          <a
            className="refund-policy external-icon"
            href="https://openstax.secure.force.com/help/articles/FAQ/OpenStax-Tutor-Beta-Student-Refund-Policy"
            tabIndex={0}
            target="_blank"
          >
            Refund policy for OpenStax Tutor Beta courses
          </a>
          <div
            className="help"
          >
            Need help? 
            <a
              href="mailto:support@openstax.org"
            >
              Contact Support
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="visible-when-debugging"
    >
      <div
        className="tour-spy-info"
      >
        available regions: []; region tour ids: []; audience tags: []; tour tags: []; eligible tours: []; TOUR RIDE: &lt;none&gt;
      </div>
    </div>
  </div>
  <a
    aria-hidden="true"
    className="debug-toggle-link"
    href="#spy"
    onClick={[Function]}
    tabIndex={-1}
  >
    π
  </a>
</div>
`;
